ros2_generate_interface_docs/ros2_generate_interface_docs/main.py

```diff
@@ -100,6 +101,11 @@ def generate_interfaces(interfaces, html_dir, template, interface_type, timestam
                 documentation_data['type'] = 'Service'
                 function_to_generate_text_from_spec = srv_utils.generate_msg_text_from_spec
 
+            if interface_type == 'action':
```

Comment:
What would you think of file constants that map interface type to the variables you use below:

So above line 32 you'd have something like:
```
DATA_TYPE_NAMES = {'msg': 'Message', 'srv': 'Service', 'action': 'Action'}
```
Then all three of these if statements can collapse into:
```
documentation_data['ext'] = interface_type
documentation_data['type'] = DATA_TYPE_NAMES[interface_type]
function_to_generate... = MESSAGE_GENERATION_FUNCTIONS[interface_type]
```

Reply:
I will include this in a follow up PR